packages/cursorless-engine/src/core/commandVersionUpgrades/canonicalizeAndValidateCommand.ts

```diff
@@ -32,26 +33,21 @@ export function canonicalizeAndValidateCommand(
   command: Command,
 ): EnforceUndefined<CommandComplete> {
   const commandUpgraded = upgradeCommand(command);
-  const {
-    action,
-    targets: inputPartialTargets,
-    usePrePhraseSnapshot = false,
-    spokenForm,
-  } = commandUpgraded;
+  const { action, usePrePhraseSnapshot = false, spokenForm } = commandUpgraded;
 
   const actionName = canonicalizeActionName(action.name);
-  const partialTargets = canonicalizeTargets(inputPartialTargets);
+  const partialTargets = getPartialTargetDescriptors(commandUpgraded);
 
+  canonicalizeTargets(partialTargets);
   validateCommand(actionName, partialTargets);
 
   return {
     version: LATEST_VERSION,
     spokenForm,
     action: {
+      ...action,
       name: actionName,
-      args: action.args ?? [],
-    },
-    targets: partialTargets,
+    } as PartialActionDescriptor,
```

Comment:
Not a fan of having to cast like this, but you to the typed nature of our actions it will be hard to get around.

Reply:
I'll take a crack at it